### PR TITLE
morethantvapi: add torznab xml support (#13118)

### PR DESCRIPTION
**Pull request by missingfile**

### CODE REVIEW

File: src/Jackett.Common/Indexers/Feeds/MoreThanTVAPI.cs [hunk 1]
```diff
@@ -59,13 +58,15 @@ public class MoreThanTVAPI : BaseNewznabIndexer
             AddCategoryMapping(TorznabCatType.MoviesHD.ID, TorznabCatType.MoviesHD);
             AddCategoryMapping(TorznabCatType.MoviesUHD.ID, TorznabCatType.MoviesUHD);
             AddCategoryMapping(TorznabCatType.MoviesBluRay.ID, TorznabCatType.MoviesBluRay);
+
+            configData.AddDynamic("keyInfo", new DisplayInfoConfigurationItem(String.Empty, "Find or Generate a new API Key by accessing your <a href=\"https://www.morethantv.me/\" target =_blank>MoreThanTV</a> account <i>User Security</i> page and scrolling to the <b>API Keys</b> section."));
```
Comment (missingfile) [inline src/Jackett.Common/Indexers/Feeds/MoreThanTVAPI.cs:62]: Please use https://www.morethantv.me/user/security

File: src/Jackett.Common/Indexers/Feeds/MoreThanTVAPI.cs [hunk 2]
```diff
@@ -59,13 +58,15 @@ public class MoreThanTVAPI : BaseNewznabIndexer
...
         }
 
         public override Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
         {
             LoadValuesFromJson(configJson);
 
-            if (configData.Passkey.Value.Length != 32)
+            if (configData.Key.Value.Length != 32)
                 throw new Exception("Invalid Passkey configured. Expected length: 32");
```
Comment (missingfile) [inline src/Jackett.Common/Indexers/Feeds/MoreThanTVAPI.cs:70]: s/Passkey/API key/
😂

**Comment by garfield69** [comment]
seem to have a mismatch in caps
from Mtv direct https://www.morethantv.me/api/torznab?t=caps
```xml
    <searching>
        <search available="yes" supportedParams="q,imdbid,tvdbid,tag" />
        <tv-search available="yes" supportedParams="q,tvdbid,season,ep,imdbid,tag" />
        <movie-search available="yes" supportedParams="q,imdbid,tag" />
    </searching>
```
from jackett http://127.0.0.1:9117/api/v2.0/indexers/morethantv-api/results/torznab/api?t=caps&apikey=redacted
```xml
  <searching>
    <search available="yes" supportedParams="q" />
    <tv-search available="yes" supportedParams="q,season,ep,tvdbid" />
    <movie-search available="yes" supportedParams="q,imdbid,tmdbid" />
    <music-search available="no" supportedParams="q" />
    <audio-search available="no" supportedParams="q" />
    <book-search available="no" supportedParams="q" />
  </searching>
```

Ignoring tag which jackett obviously does not support, mtv has tv-search with imdbid and movie-search without tmdbid
while mtv may add tmdbid to their movie-search eventually, we should probably drop tmdbid on ours for now, and add imdbid to tvsearch ?

**Comment by missingfile** [comment]
Sounds good. tmdbid is very weird because it only makes sense to use with t=tvsearch or t=movie and NOT t=search. The reason is TMDB ID is an integer that can be reused for both movie and/or TV.

**Comment by missingfile** [comment]
FYI, IMDb search is disabled globally for TV in Jackett per https://github.com/Jackett/Jackett/issues/8107.
I think we want to keep it here so it'll magically work when that issue is resolved.

**Comment by garfield69** [comment]
how can we pick up the last category retuned by the xml, so that we can tag the more specific cat?
eg. mtv api returns 
```xml
    <category>5000</category>
    <category>5040</category>
```
and we would like to use the 5040 to id the category in the jackett dashboard search results page

**Review by garfield69 — APPROVED** [review]
Thank you very much for coding this C# indexer.

**Pull request merged**